load_cart.php

```diff
@@ -1,64 +1,86 @@
 <?php
 session_start();
-include 'connectdb.php';  // Reuse existing database connection
+
+$userType = $_SESSION['type'];
 
 if (!isset($_SESSION['order_id'])) {
     die('Order ID not set in session.');
 }
-
 $order_id = $_SESSION['order_id'];
 
-try {
-    // Fetch order details with dish information in a single query
-    $sql = "SELECT od.*, m.dish_name, m.image 
-            FROM order_detail od 
-            JOIN menu m ON od.dish_id = m.dish_id 
-            WHERE od.order_id = :order_id AND od.status = 'inprogress'";
+// Database connection
+$conn = new mysqli('localhost', 'root', '', 'rms');
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider using config for DB credentials</b></div>
<div id="fix">

Consider using prepared statements consistently. The database credentials are hardcoded which could pose a security risk. Consider moving these to a configuration file.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
require_once 'config.php';
$conn = new mysqli($db_host, $db_user, $db_pass, $db_name);
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Lux-Jason/rms_new/pull/12#issuecomment-2561647433>#75a1e4</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged